### PR TITLE
store: use the "publisher" when populating the "publisher" field 

### PR DESCRIPTION
**Pull request by mvo5**
The store provides publisher information now. We should use it. This will fix that e.g. `snap info aws-cli` shows: "publisher: aws" instead of "publsiher: Amazaon Web Services, Inc.".

This fixes https://bugs.launchpad.net/ubuntu/+source/gnome-software/+bug/1742637

### CODE REVIEW

**Comment by mvo5** [comment]
@chipaca I am not sure what we need to do next. Probably something to expose both developer and publisher via the REST API. This is just a first fix to make the client be more correct. Happy to hear your thoughts on this, I was thinking we provide "developer" and "publisher" (and there ids) via the rest api.

**Comment by chipaca** [comment]
@mvo5 me either. This is fine for now (hence my +1) but there's more to be done I'm sure.

File: store/store_test.go [hunk 1]
```diff
@@ -2003,7 +2003,7 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryDetails(c *C) {
 	c.Check(result.Architectures, DeepEquals, []string{"all"})
 	c.Check(result.Revision, Equals, snap.R(27))
 	c.Check(result.SnapID, Equals, helloWorldSnapID)
-	c.Check(result.Publisher, Equals, "canonical")
+	c.Check(result.Publisher, Equals, "Canonical")
```
Comment (zyga) [inline store/store_test.go]: This broke tests:
```
in test-snapd-python-webserver.publisher expected 'canonical', got 'Canonical'
```

Comment (bboozzoo) [reply store/store_test.go]: Actually it's https://github.com/snapcore/snapd/pull/4532/files#diff-335c37cf601209543ed44f0aeca1073aR94 that broke

File: tests/main/snap-info/check.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def verRevNotesRx(s):
 
 check("test-snapd-devmode", res[3],
    ("name", equals, "test-snapd-devmode"),
-   ("publisher", equals, "canonical"),
+   ("publisher", equals, "Canonical"),
```
Comment (bboozzoo) [inline tests/main/snap-info/check.py]:
```
name:      test-snapd-devmode
summary:   Basic snap with devmode confinement
publisher: canonical
contact:   snappy-canonical-storeaccount@canonical.com
description: |
  A basic buildable snap that asks for devmode confinement
snap-id:     821MII7GAzoRnPvTEb8R51Z1s9e0XmK5
channels:            
  stable:    –           
  candidate: –           
  beta:      1.0 (1) 4kB devmode
  edge:      1.0 (1) 4kB devmode
```

**Pull request merged**